src/lib/ext-activitylog.js

```diff
@@ -94,6 +148,13 @@ class Controller {
       this.view.setError(error.message);
     }
   }
+
+  onFilterChange(filterObject) {
+    const { filterDetail, isFilterRemoved } = filterObject;
+
+    this.model[isFilterRemoved ? 'removeFilter' : 'addFilter'](filterDetail);
+    this.view.handleTableRows({ newLogs: null, filterObj: this.model.filter });
```

Comment:
@Rob--W I could pass the (null, this.model.filter) as argument with `handleTableRows` method, but I have put the values in an object with meaning full keys instead of passing them as an argument. Does this improve the readability ?

Reply:
So this is a follow-up to my review comment at https://github.com/mozilla/extension-activity-monitor/pull/10#discussion_r456694760

The issue is that you are still having two code paths that currently pass `this.model.filter`, but without any guarantee within the table-view-element that each row consistently uses the same filter.

Also, in the `handleTableRows` method, you're currently doing:

```
if (newLogs) {
  // 1: appends new log
}
// 2: update filters for all logs
```

The second part is very wasteful, because it recomputes the filter result for all logs, even if all you're doing is adding a new log.

The filter only needs to be updated on `filterchange`, and at initialization. Currently you're tying the model assignment to the row creation (append log), which works, but is not ideal. It will have maintainability issues (see start of comment) and performance issues (see previous paragraph).

To address all of the above issues, I suggest to create a separate method to set the filter, e.g. `setLogFilter` that takes a callback. The callback can be stored in the log-table-view, and is passed the log and returns a boolean. This is a very clear interface. This also relates to my comment about arrow functions at https://github.com/mozilla/extension-activity-monitor/pull/10#discussion_r456690853